Aws api gateway documentation part (#849)
inspec/inspec-aws: docs/resources/aws_api_gateway_documentation_part.md

---
title: About the aws_api_gateway_documentation_part Resource
platform: aws
---

# aws_api_gateway_documentation_part

> Use the `aws_api_gateway_documentation_part` InSpec audit resource to test properties of a single AWS ApiGateway Deployment.

Reviewer: ```suggestion
Use the `aws_api_gateway_documentation_part` InSpec audit resource to test properties of a single AWS ApiGateway DocumentationPart.
```

> The AWS::ApiGateway::Deployment resource deploys an API Gateway RestApi resource to a stage so that clients can call the API over the internet. The stage acts as an environment.

Reviewer: ```suggestion
The AWS::ApiGateway::DocumentationPart resource creates a documentation part for an API.
```

| location.type        | The type of API entity to which the documentation content applies.                |
| location.path        | The URL path of the target.                                                       |
| location.method      | The HTTP verb of a method.                                                        |
| location.status_code | The HTTP status code of a response.                                               |
| location.name        | The name of the targeted API entity.                                              |
| properties           | A content map of API-specific key-value pairs describing the targeted API entity. |

## Examples

### Ensure that the documentation part ID exists.

    describe aws_api_gateway_documentation_part(rest_api_id: 'REST_API_ID', documentation_part_id: 'DOCUMENTATION_PART_ID') do
      its('id') { should eq 'DOCUMENTATION_PART_ID' }
    end

### Check whether the API path exist for a documentation part.

    describe aws_api_gateway_documentation_part(rest_api_id: 'REST_API_ID', documentation_part_id: 'DOCUMENTATION_PART_ID') do

>         its('location.path') { should eq REQUEST_BODY }

Reviewer: ```suggestion
        its('location.path') { should eq 'REQUEST_BODY' }
```
